vllm/core/block_manager_v2.py

```diff
@@ -467,9 +467,24 @@ def _can_swap(self,
             AllocStatus: The AllocStatus for swapping in/out the given 
                 sequence_group on to the 'device'.
         """
-        blocks = self._get_blocks_for_swap(seq_group, status)
+        # First determine the number of blocks that will be touched to swap the 
+        # already full blocks.
+        blocks = []
+        for seq in seq_group.get_seqs(status=status):
+            block_table = self.block_tables[seq.seq_id]
+            if block_table.blocks is not None:
+                for block in block_table.blocks:
+                    if block.is_full:
+                       blocks.append(block)
```

Comment:
1. We could directly fix `_get_blocks_for_swap`.
2. Please also add comment to the docstring in `_get_blocks_for_swap` about we can only swap out full blocks.

Reply:
I am retaining this logic here because in this loop I also compute the list of unseen tokens for the sequences. PTAL if this is fine